dictdiffer/__init__.py

```diff
@@ -134,126 +140,133 @@ def dotted(node, default_type=list):
-        len_first = len(first)
-        len_second = len(second)
-
-        intersection = list(range(0, min(len_first, len_second)))
-        addition = list(range(min(len_first, len_second), len_second))
-        deletion = list(reversed(range(min(len_first, len_second), len_first)))
-
-        differ = True
-
-    elif isinstance(first, SET_TYPES) and isinstance(second, SET_TYPES):
-        # Deep copy is not necessary for hashable items.
-        addition = second - first
-        if len(addition):
-            yield ADD, dotted_node, [(0, addition)]
-        deletion = first - second
-        if len(deletion):
-            yield REMOVE, dotted_node, [(0, deletion)]
-
-    if differ:
-        # Compare if object is a dictionary or list.
-        #
-        # NOTE variables: intersection, addition, deletion contain only
-        # hashable types, hence they do not need to be deepcopied.
-        #
-        # Call again the parent function as recursive if dictionary have child
-        # objects.  Yields `add` and `remove` flags.
-        for key in intersection:
-            # if type is not changed, callees again diff function to compare.
-            # otherwise, the change will be handled as `change` flag.
-            if path_limit and path_limit.path_is_limit(node+[key]):
-                yield CHANGE, node+[key], (
-                    deepcopy(first[key]), deepcopy(second[key])
-                )
-            else:
-                recurred = diff(first[key],
-                                second[key],
-                                node=node + [key],
-                                ignore=ignore,
-                                path_limit=path_limit,
-                                expand=expand,
-                                tolerance=tolerance)
-
-                for diffed in recurred:
-                    yield diffed
-
-        if addition:
-            if path_limit:
-                collect = []
-                collect_recurred = []
-                for key in addition:
-                    if not isinstance(second[key],
-                                      SET_TYPES + LIST_TYPES + DICT_TYPES):
-                        collect.append((key, deepcopy(second[key])))
-                    elif path_limit.path_is_limit(node+[key]):
-                        collect.append((key, deepcopy(second[key])))
-                    else:
-                        collect.append((key, second[key].__class__()))
-                        recurred = diff(second[key].__class__(),
-                                        second[key],
-                                        node=node+[key],
-                                        ignore=ignore,
-                                        path_limit=path_limit,
-                                        expand=expand,
-                                        tolerance=tolerance)
+    def _diff_recurrence(_first, _second, _node=None):
```

Comment:
maybe a better name would be `_diff_recursive` since it is called it from its body.

Reply:
If u want, no pb for me